File: functions/a_stock_moneyflow.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
A 股板块资金流 - Netlify Function
定时执行：每个交易日 15:30 (UTC+8)
"""

import json
import akshare as ak
from datetime import datetime
from typing import List, Dict
import pandas as pd


def get_sector_flow() -> List[Dict]:
    """获取 A 股板块资金流数据"""
    try:
        df = ak.stock_fund_flow_industry(symbol="即时")
        
        sectors = []
        for _, row in df.iterrows():
            sectors.append({
                'rank': int(row['序号']) if pd.notna(row['序号']) else 0,
                'name': str(row['行业']),
                'index': float(row['行业指数']) if pd.notna(row['行业指数']) else 0,
                'change_pct': float(row['行业 - 涨跌幅']) if pd.notna(row['行业 - 涨跌幅']) else 0,
                'inflow': float(row['流入资金']) if pd.notna(row['流入资金']) else 0,
                'outflow': float(row['流出资金']) if pd.notna(row['流出资金']) else 0,
                'net_money': float(row['净额']) if pd.notna(row['净额']) else 0,
                'company_count': int(row['公司家数']) if pd.notna(row['公司家数']) else 0,
            })
        
        return sectors
    except Exception as e:
        print(f"获取数据失败：{e}")
        return []


def handler(event, context):
    """Netlify Function 入口"""
    print(f"执行时间：{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
    
    # 获取数据
    sectors = get_sector_flow()
    
    if not sectors:
        return {
            "statusCode": 500,
            "body": json.dumps({"error": "获取数据失败"}, ensure_ascii=False)
        }
    
    # 按净流入排序
    sectors_sorted = sorted(sectors, key=lambda x: x['net_money'], reverse=True)
    
    # 生成排名数据
    ranking = []
    for i, s in enumerate(sectors_sorted, 1):
        ranking.append({
            'rank': i,
            'name': s['name'],
            'net_money': round(s['net_money'], 2),
            'inflow': round(s['inflow'], 2),
            'outflow': round(s['outflow'], 2),
            'change_pct': round(s['change_pct'], 2)
        })
    
    # 生成输出
    output = {
        'date': datetime.now().strftime('%Y-%m-%d %H:%M'),
        'market': 'A 股',
        'ranking': ranking
    }
    
    # 返回 JSON
    return {
        "statusCode": 200,
        "body": json.dumps(output, ensure_ascii=False, indent=2),
        "headers": {
            "Content-Type": "application/json; charset=utf-8"
        }
    }


if __name__ == "__main__":
    # 本地测试
    result = handler({}, {})
    print(result["body"])

File: functions/a_stock_moneyflow.js
const axios = require('axios');

exports.handler = async (event, context) => {
  try {
    // 东方财富行业板块资金流 - 提供收盘后数据
    const response = await axios.get('http://push2.eastmoney.com/api/qt/clist/get', {
      params: {
        pn: 1,
        pz: 50,
        po: 1,
        np: 1,
        ut: 'bd1d9ddb04089700cf9c27f6f7426281',
        fltt: 2,
        invt: 2,
        fid: 'f62',
        fs: 'm:90 t:3',  // 行业板块
        fields: 'f12,f14,f62,f66,f124'
      },
      headers: {
        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36',
        'Referer': 'http://quote.eastmoney.com/'
      }
    });

    console.log('A 股 Response:', response.data);
    
    if (!response.data || !response.data.data || !response.data.data.diff) {
      return {
        statusCode: 200,
        body: JSON.stringify({ 
          date: new Date().toISOString(), 
          market: 'A 股', 
          ranking: [],
          message: '暂无数据'
        }, null, 2)
      };
    }

    const data = response.data.data.diff;
    
    // 按净流入排序
    const sorted = data.sort((a, b) => (parseFloat(b.f62)||0) - (parseFloat(a.f62)||0));

    // 东方财富数据单位是万，直接使用
    const ranking = sorted.map((item, i) => ({
      rank: i + 1,
      name: item.f14 || '',  // 板块名称
      net_money: parseFloat(item.f62) || 0,  // 净流入（万）
      change_pct: parseFloat(item.f124) || 0  // 涨跌幅
    }));

    return {
      statusCode: 200,
      body: JSON.stringify({ 
        date: new Date().toISOString(), 
        market: 'A 股', 
        ranking,
        unit: '万元'
      }, null, 2),
      headers: {
        'Content-Type': 'application/json; charset=utf-8',
        'Access-Control-Allow-Origin': '*'
      }
    };
  } catch (error) {
    console.error('A 股 Error:', error);
    return { 
      statusCode: 500, 
      body: JSON.stringify({ error: error.message }),
      headers: { 'Content-Type': 'application/json' }
    };
  }
};
